src/include/daos_obj_class.h

```diff
@@ -467,6 +467,17 @@ struct daos_oclass_list {
 int
 daos_oclass_name2id(const char *name);
 
+/**
+ * Return the list of object class.
+ *
+ * \param[in]	size	length in bytes of str buffer.
+ * \param[out]	str	buffer to get all registered oclass names
+ *
+ * \return		>= 0 on success and length of str, -1 if error.
```

Comment:
looking at the implementation it returns the required length of the str - that may exceed size bytes. i.e., the function doesn't return strlen(str) after it fills in as many object class names as it can. Can this description of the return value be modified to match the behavior?

Reply:
I have changed the description in new commit